subprojects/kotlin-dsl/src/main/kotlin/org/gradle/kotlin/dsl/support/IO.kt

```diff
@@ -39,3 +39,11 @@ fun Appendable.appendReproducibleNewLine(value: CharSequence = ""): Appendable =
 internal
 fun File.isParentOf(child: File): Boolean =
     child.canonicalPath.startsWith(canonicalPath)
+
+
+/**
+ * List files ordered by filename for reproducibility.
+ * Never returns null.
+ */
+fun File.listFilesOrdered(filter: (File) -> Boolean = { true }): List<File> =
+    listFiles()?.filter(filter)?.sortedBy { it.name } ?: emptyList()
```

Comment:
⭕  These collections are probably very small but it still bothers me a little that we would call `filter(filter)` when there's no filtering to be done. How about a nullable filter and something like `listFiles()?.let { if (filter != null) it.filter(filter) else it }?.sortedBy ... `